streamalert_cli/athena/helpers.py

```diff
@@ -57,10 +64,21 @@ def add_partition_statement(partitions, bucket, table_name):
             continue
 
         fmt_values.update(parts.groupdict())
+        partition_stmt = PARTITION_STMT.format(**fmt_values)
+
+        # If statement will be reset if it is about to reach the MAX_QUERY_LENGTH
+        if statement_len - len(partition_stmt) - len(statement) < 0:
```

Comment:
store the value of `len(partition_stmt)`  to avoid calculating twice (L75 below)